.github/workflows/amplify_integration_tests.yaml

```diff
@@ -0,0 +1,108 @@
+        with:
+          distribution: 'corretto' # Amazon Corretto Build of OpenJDK
+          java-version: '11'
+      
+      - name: Install dependencies
+        uses: ./.github/composite_actions/install_dependencies
+
+      - name: Fetch Amplify backend configurations
+        uses: ./.github/composite_actions/fetch_backends
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          scope: ${{ matrix.scope }}
+          api-app-id: ${{ secrets.API_APP_ID }}
+          auth-app-id: ${{ secrets.AUTH_APP_ID }}
+          datastore-app-id: ${{ secrets.DATASTORE_APP_ID }}
+          storage-app-id: ${{ secrets.STORAGE_APP_ID }}
+
+      - name: Build example app with integration tests
+        run: |
+          melos exec --scope=${{ matrix.scope }} --file-exists="integration_test/main_test.dart" --fail-fast flutter build apk --verbose
+
+      - name: Run Android integration tests
+        uses: reactivecircus/android-emulator-runner@76c2bf6f95ed6458fd659a1bdb680a0f8df232dc # 2.23.0
+        timeout-minutes: 25
+        with:
+          api-level: 29
+          # retry once
+          script: melos exec --scope=${{ matrix.scope }} "flutter test --no-pub integration_test/main_test.dart -d emulator-5554" || melos exec --scope=${{ matrix.scope }} "flutter test --no-pub integration_test/main_test.dart -d emulator-5554"
+
+  ios:
+    runs-on: macos-latest
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        scope: ["amplify_api_example"] #, "amplify_storage_s3_example", "amplify_auth_cognito_example"]
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # 2.4.0
+        with:
+          persist-credentials: false
+      
+      - name: Install dependencies
+        uses: ./.github/composite_actions/install_dependencies
+
+      - name: Boot iOS Simulator
+        run: |
+          xcrun simctl boot "iPhone 12"
+
+      - name: Fetch Amplify backend configurations
+        uses: ./.github/composite_actions/fetch_backends
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          scope: ${{ matrix.scope }}
+          api-app-id: ${{ secrets.API_APP_ID }}
+          auth-app-id: ${{ secrets.AUTH_APP_ID }}
+          datastore-app-id: ${{ secrets.DATASTORE_APP_ID }}
+          storage-app-id: ${{ secrets.STORAGE_APP_ID }}
+
+      - name: Build example app with integration tests
+        run: |
+          melos exec --scope=${{ matrix.scope }} flutter build ios --simulator
+
+      - name: Run integration tests
+        # retry once
+        id: iosTestFirstAttempt
+        continue-on-error: true
+        timeout-minutes: 10
+        run: |
+          melos exec --scope=${{ matrix.scope }} "flutter test --no-pub integration_test/main_test.dart -d iPhone"
```

Comment:
Same comment as above - can we update integ script if needed so it can be used here?

Reply:
done